File: frontend/src/types/contratto.ts
interface Contratto {
  id: number;
  cliente_id: number;
  tipo_contratto_id: number;
  data_inizio: string;
  data_fine: string | null;
  stato: "attivo" | "scaduto" | "sospeso";
  note: string | null;
  created_at: string;
  updated_at: string;
}

interface ContrattoCreate {
  cliente_id: number;
  tipo_contratto_id: number;
  data_inizio: string;
  data_fine?: string;
  stato?: "attivo" | "scaduto" | "sospeso";
  note?: string;
}

type ContrattoUpdate = Partial<ContrattoCreate>;

export type { Contratto, ContrattoCreate, ContrattoUpdate };

File: backend/app/schemas/contratto.py
"""Contratto-related Pydantic schemas for request/response validation."""
from datetime import datetime, date
from typing import Optional

from pydantic import BaseModel, ConfigDict


class ContrattoBase(BaseModel):
    """Base schema for Contratto with common fields."""
    cliente_id: int
    tipo_contratto_id: int
    data_inizio: date
    data_fine: Optional[date] = None
    stato: str = "attivo"
    note: Optional[str] = None


class ContrattoCreate(ContrattoBase):
    """Schema for contratto creation request."""
    pass


class ContrattoUpdate(BaseModel):
    """Schema for contratto update request with all optional fields."""
    cliente_id: Optional[int] = None
    tipo_contratto_id: Optional[int] = None
    data_inizio: Optional[date] = None
    data_fine: Optional[date] = None
    stato: Optional[str] = None
    note: Optional[str] = None


class ContrattoResponse(ContrattoBase):
    """Schema for contratto response data."""
    model_config = ConfigDict(from_attributes=True)

    id: int
    created_at: datetime
    updated_at: datetime